### PR TITLE
fix(zarrmultiscalespatialimage): use a promise queue to manage resolving chunk promises

### PR DESCRIPTION
**Pull request by bnmajor**
Prevents the network from getting overwhelmed trying to handle all of the promises at once for higher resolution levels. For larger data there may be quite a few chunks and if overwhelmed the data may not load the full resolution.

### CODE REVIEW

**Comment by bnmajor** [comment]
@PaulHax FYI the promise queue ended up being the solution, thanks for your help!

**Review by thewtex — COMMENTED** [review]
wOOt!

File: src/IO/ZarrMultiscaleSpatialImage.js [hunk 1]
```diff
@@ -204,6 +206,7 @@ class ZarrMultiscaleSpatialImage extends MultiscaleSpatialImage {
     const chunkPathBase = info.pixelArrayPath
     const chunkPaths = []
     const chunkPromises = []
+    const chunksQueue = new PQueue({ concurrency: 5 })
```
Comment (thewtex) [inline src/IO/ZarrMultiscaleSpatialImage.js:209]: Can we bump this up to something higher, e.g. `10`, `16`, `32`?

Comment (bnmajor) [reply src/IO/ZarrMultiscaleSpatialImage.js]: Oof, yes, that was a mistake. I actually had it at 10 but some more testing looks like we can set it to 16 with consistent results. Values above that are inconsistent. Fixed to be 16 now.

**Review by thewtex — APPROVED** [review]

**Pull request merged**